backend/src/lib/search.ts

```diff
@@ -162,3 +159,34 @@ export function getSearchResults(
     req.end()
   })
 }
+
+export async function pingSearchAPI() {
+  const options = await getServiceAccountOptions()
+  return new Promise<boolean>((resolve, reject) => {
+    let body = ''
+    const req = request(options, (res) => {
+      res.on('data', (data) => {
+        body += data
+      })
+      res.on('end', () => {
+        try {
+          const result = JSON.parse(body) as { data: unknown }
+          if (result.data) {
+            resolve(true)
+          } else {
+            reject('nope')
+          }
+        } catch (e) {
+          reject(e)
+        }
+      })
+    })
+    req.on('error', (e) => {
+      reject(e)
+    })
+    req.write(
+      '{"operationName":"searchResult","variables":{"input":[{"filters":[{"property":"kind","values":["Application"]},{"property":"apigroup","values":["argoproj.io"]},{"property":"cluster","values":["!local-cluster"]},{"property":"name","values":["a*","i*","n*"]}],"limit":1}]},"query":"query searchResult($input: [SearchInput]) {\\n  searchResult: search(input: $input) {\\n    items\\n  }\\n}"}'
+    )
```

Comment:
This seems like a rather arbitrary query to execute as a test. Perhaps @jlpadilla or @zlayne might have a suggestion for a simpler check for whether the search API is ready.

Reply:
using @zlayne suggestion to ping for existence of pod